game-core/src/main/java/games/strategy/engine/data/GameMap.java

```diff
@@ -207,6 +209,46 @@ public Territory getTerritory(final String s) {
     return getNeighbors(frontier, searched, distance, null);
   }
 
+  /**
+   * Returns all neighbors within a certain distance of the starting territory that territories
+   * between the 2 match the condition. Does NOT include the original/starting territory in the
+   * returned Set.
+   */
+  public Set<Territory> getNeighborsIgnoreEnd(final Territory territory, final int distance,
+      final Predicate<Territory> cond) {
+    if (distance < 0) {
+      throw new IllegalArgumentException("Distance must be positive not: " + distance);
+    }
+    if (distance == 0) {
+      return Collections.emptySet();
+    }
+    final Set<Territory> neighbors = new HashSet<>(getNeighbors(territory));
+    if (distance == 1) {
+      return neighbors;
+    }
+    final Set<Territory> start = getNeighbors(territory, cond);
+    for (int i = 2; i <= distance; i++) {
```

Comment:
Minor: I wonder a bit why the value '2'. Extracting it to a constant and adding a comment on it could be a good way to document the reasoning why the number '2' is here. I would guess it is so we do not consider too many territories, but not sure if maybe the reason is because we can assume movement is limited.

Reply:
You start with 2 since there are specific conditions already for <0, 0, and 1. So if distance is 2 or more than you start doing the recursion calls for each distance. I don't think a constant would really make sense here.

Reply:
Okay, I wonder a bit why not go up to 3 or higher, but I'll take this at face-value. :+1: